File: Core/src/main/java/com/github/ezauton/core/simulation/ActionScheduler.java
package com.github.ezauton.core.simulation;

import com.github.ezauton.core.action.Action;

import java.util.concurrent.Future;

/**
 * An interface which is used to schedule an action in a certain way. Nice for simulations.
 */
public interface ActionScheduler {
    /**
     * @param action
     * @return the action
     */
    Future<Void> scheduleAction(Action action);
}

File: Core/src/main/kotlin/com/github/ezauton/core/simulation/ActionScheduler.kt
package com.github.ezauton.core.simulation

import com.github.ezauton.core.action.Action
import kotlinx.coroutines.Deferred

/**
 * An interface which is used to schedule an action in a certain way. Nice for simulations.
 */
interface ActionScheduler {
  /**
   * @param action
   * @return the action
   */
  fun <T> scheduleAction(action: Action<T>): Deferred<T>
}
